gems/aws-sdk-cognitoidentity/lib/aws-sdk-cognitoidentity/customizations/cognito_identity_credentials.rb

```diff
@@ -0,0 +1,102 @@
+require 'set'
+require 'securerandom'
+require 'base64'
+
+module Aws
+  module CognitoIdentity
+    # An auto-refreshing credential provider
+    # TODO: Add good documentation....
+    class CognitoIdentityCredentials
+
+      include CredentialProvider
+      include RefreshingCredentials
+
+      # @option options [String] :identity_id the Cognito identity_id.  Required
+      # unless identity_pool_id is given. A unique
+      # identifier in the format REGION:GUID
+      #
+      # @option options [String] :identity_pool_id Required unless identity_id is
+      # provided. An identity pool ID in the format REGION:GUID.
+      #
+      # @option options [Hash<String,String>] :logins A set of optional
+      # name-value pairs that map provider names to provider tokens.
+      # The name-value pair will follow the syntax
+      # "provider_name": "provider_user_identifier".
```

Comment:
Do some general formatting here (and in other places). Indent two spaces for each additional line on the same documentation line. (i.e. two spaces at name-values, etc)